webapp/tests/forms/steps/test_steuerlotse_step.py

```diff
@@ -533,89 +438,38 @@ def test_update_session_data_is_called(self):
+                                            next_step=MockRenderStep)
+            mock_yesno_step.handle()
+            data_after_second_handle = mock_yesno_step.stored_data
 
-
-class TestSteuerlotseFormStepOverrideSessionData(unittest.TestCase):
-
-    def test_data_is_saved_to_empty_session(self):
-        new_data = {'brother': 'Luigi'}
-        with app.app_context() and app.test_request_context() as req:
-            with patch('app.forms.steps.steuerlotse_step.serialize_session_data', MagicMock(side_effect=lambda _: _)):
-                self.assertNotIn('form_data', req.session)
-                MockFormStep(endpoint="lotse")._override_session_data(new_data)
-                self.assertIn('form_data', req.session)
-                self.assertEqual(new_data, req.session['form_data'])
-
-    def test_data_is_saved_to_prefilled_session(self):
-        new_data = {'brother': 'Luigi'}
-        with app.app_context() and app.test_request_context() as req:
-            with patch('app.forms.steps.steuerlotse_step.serialize_session_data', MagicMock(side_effect=lambda _: _)):
-                req.session = {'form_data': {'brother': 'Mario', 'pet': 'Yoshi'}}
-                self.assertIn('form_data', req.session)
-                MockFormStep(endpoint="lotse")._override_session_data(new_data)
-                self.assertIn('form_data', req.session)
-                self.assertEqual(new_data, req.session['form_data'])
-
-    def test_if_data_stored_with_other_identifier_then_it_is_not_changed(self):
-        new_data = {'brother': 'Luigi'}
-        other_data = {'enemy': 'Bowser'}
-        with app.app_context() and app.test_request_context() as req:
-            with patch('app.forms.steps.steuerlotse_step.serialize_session_data', MagicMock(side_effect=lambda _: _)):
-                req.session = {'form_data': {'brother': 'Mario', 'pet': 'Yoshi'},
-                               'OTHER_IDENTIFIER': other_data}
-                MockFormStep(endpoint="lotse")._override_session_data(new_data)
-                self.assertEqual(other_data, req.session['OTHER_IDENTIFIER'])
-
-    def test_if_stored_data_identifier_is_set_then_override_session_data_with_that_new_identifier(self):
-        new_data = {'brother': 'Luigi'}
-        other_data = {'enemy': 'Bowser'}
-        new_identifier = "NEW_IDENTIFIER"
-        with app.app_context() and app.test_request_context() as req:
-            with patch('app.forms.steps.steuerlotse_step.serialize_session_data', MagicMock(side_effect=lambda _: _)):
-                req.session = {'form_data': {'brother': 'Mario', 'pet': 'Yoshi'},
-                               'OTHER_IDENTIFIER': other_data}
-                step = MockFormStep(endpoint="lotse")
-                step.session_data_identifier = new_identifier
-                step._override_session_data(new_data)
-                self.assertEqual(new_data, req.session[new_identifier])
+        self.assertEqual({'yes_no_field': None}, data_after_second_handle)
```

Comment:
I understand why you did it that way. However, this test is an older test that we precisely did to test the data in the session. So, I think we should still deserialize the session and let handle handle all of that.

Potentially, we need to move those tests to the stepchooser tests, but I do not know.

Reply:
Are you sure that we didn't want to test the data in general? I think this rather tests the (rather complicated) `process()` method in the `YesNoField`. And there the important thing is that the field's id is correctly set to None in the data, not necessarily in the session (of course also in the session, but we test that the session is overriden by the data in stored_data some other place.)